### ISSUE TITLE
[BUG] OHIT ignores random_state (uses global np.random)

### ISSUE THREAD

**Issue by TonyBagnall** [aeon-toolkit/aeon#3501]
### Describe the bug

_ohit.py:276: _generate_synthetic_samples calls np.random.multivariate_normal(...) on the global RNG, despite _transform creating a seeded random_state at line 139. OHIT is therefore non-deterministic even with a fixed seed.

Fix: pass the check_random_state generator into _generate_synthetic_samples and use rng.multivariate_normal(...); also build it once in _fit from self.random_state rather than re-seeding inside _transform. Add a reproducibility test asserting two same-seed runs are equal.

### Steps/Code to reproduce the bug

```python
r1 = OHIT(random_state=49).fit_transform(X, y)[0]
r2 = OHIT(random_state=49).fit_transform(X, y)[0]
np.array_equal(r1, r2)   # False ; max|diff| ≈ 1.36
```

### Expected results

identical output for identical random_state.

### Actual results

 outputs differ by up to 1.36.
